server/client.go

```diff
@@ -2420,13 +2417,10 @@ func (c *client) newServiceReply(tracking bool) []byte {
 	}
 	reply := b[:]
 	if tracking {
-		if c.in.shash == nil {
-			c.in.shash = c.srv.sysHash()
-		}
 		// Add in our tracking identifier. This allows the metrics to get back to only
 		// this server without needless SUBS/UNSUBS.
 		reply = append(reply, '.')
-		reply = append(reply, c.in.shash...)
+		reply = append(reply, c.srv.sys.shash...)
```

Comment:
Just to make sure... there is no chance we get here without eventsEnabled() being false?

Reply:
yes, but will add a test just because.. ;)